scripts/irods/test/test_misc.py

```diff
@@ -317,3 +319,29 @@ def test_server_correctly_cleans_up_proc_files__issue_6231(self):
         # Show that the proc directory will eventually become empty.
         lib.delayAssert(lambda: len(os.listdir(paths.proc_directory())) == 0)
 
+    def test_iinit_does_not_crash_from_long_home_directory__issue_5411(self):
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            # Create new home directory for test
+            temp_home = os.path.join(temp_dir_name, 'crazy', 'long', 'temp', 'home', 'directory',
+                                     'like', 'this', 'is', 'longer', 'than', 'the', 'solos', 'in', 'chameleon')
+            os.makedirs(temp_home)
+
+            # Setup environment
+            temp_env = os.environ.copy()
+            temp_env['HOME'] = temp_home
+
+            # Execute iinit
+            user_input = os.linesep.join(
+                ['localhost', '1247', 'rods', 'tempZone', 'rods']) + os.linesep
+            res = subprocess.run('iinit', input=user_input.encode(
+            ), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=temp_env)
+
+            self.assertEqual(res.returncode, 0)
+
+            # Implicitly check we ran through our input
+            self.assertIn(b'Enter your current iRODS password:\n', res.stdout)
```

Comment:
Consider checking for all prompts?

L334-L335 includes input for multiple prompts.